### ISSUE TITLE
Why is the `VRServiceCreator` API needed?

### ISSUE THREAD

**Issue by asajeffrey** [servo/rust-webvr#59]
Why do we expose a `VRServiceCreator` API? Why does a user write:
```rust
FooVRServiceCreator::new()::new_service()
```
rather than just:
```rust
FooVRService::new()
```

**Comment by asajeffrey** [comment]
In particular, for services that take ownership of some parameters, the `VRServiceCreator` API is not ideal:
```rust
let foo_creator = FooServiceCreator::new(this, that);
let foo1 = foo_creator.new_service(); // foo1 now owns this and that
let foo2 = foo_creator.new_service(); // What happens now? panic?
```

**Comment by jdm** [comment]
The only reason I can see for its existence is for [this code](https://github.com/servo/rust-webvr/blob/f84a20813374e44051f79c8ef88d74ca19a753d1/rust-webvr/src/vr_manager.rs#L60), but it's not clear to me that that abstraction is necessary.

**Comment by asajeffrey** [comment]
Indeed, why not keep an array of `VRService` objects rather than `VRServiceCreator` objects?

**Comment by MortimerGoro** [comment]
It's used to register services based on cargo features and in order to do lazy initialization (only initialize services when first requested from JavaScript thread, otherwise some system UI may trigger (e.g Steam VR state UI on HTC Vive).

I think we could switch to a VRService array as long as mantain a lazy initialization

**Comment by asajeffrey** [comment]
The `VRService` doesn't need to do any initialization until `initialize` is called, which won't happen until JS fetches the displays or gamepads.
